analyzer/src/main/kotlin/managers/Carthage.kt

```diff
@@ -0,0 +1,282 @@
+import org.ossreviewtoolkit.downloader.VersionControlSystem
+import org.ossreviewtoolkit.model.Hash
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.Package
+import org.ossreviewtoolkit.model.Project
+import org.ossreviewtoolkit.model.ProjectAnalyzerResult
+import org.ossreviewtoolkit.model.RemoteArtifact
+import org.ossreviewtoolkit.model.VcsInfo
+import org.ossreviewtoolkit.model.VcsType
+import org.ossreviewtoolkit.model.config.AnalyzerConfiguration
+import org.ossreviewtoolkit.model.config.RepositoryConfiguration
+import org.ossreviewtoolkit.model.jsonMapper
+import org.ossreviewtoolkit.utils.normalizeVcsUrl
+
+/**
+ * The [Carthage](https://github.com/Carthage/Carthage) package manager for Swift.
+ */
+class Carthage(
+    name: String,
+    analysisRoot: File,
+    analyzerConfig: AnalyzerConfiguration,
+    repoConfig: RepositoryConfiguration
+) : PackageManager(name, analysisRoot, analyzerConfig, repoConfig) {
+    class Factory : AbstractPackageManagerFactory<Carthage>("Carthage") {
+        // TODO: Add support for the Cartfile.
+        //  This would require to resolve the actual dependency versions as a Cartfile supports dynamic versions.
+        override val globsForDefinitionFiles = listOf("Cartfile*.resolved")
+
+        override fun create(
+            analysisRoot: File,
+            analyzerConfig: AnalyzerConfiguration,
+            repoConfig: RepositoryConfiguration
+        ) = Carthage(managerName, analysisRoot, analyzerConfig, repoConfig)
+    }
+
+    override fun resolveDependencies(definitionFile: File): List<ProjectAnalyzerResult> {
+        // Transitive dependencies are only supported if the the dependency itself uses Carthage.
+        // See: https://github.com/Carthage/Carthage#nested-dependencies
+        val workingDir = definitionFile.parentFile
+        val (user, project, ref) = getProjectInfoFromVcs(workingDir)
+        return listOf(
+            ProjectAnalyzerResult(
+                project = Project(
+                    id = Identifier(
+                        type = managerName,
+                        namespace = user.orEmpty(),
+                        name = project.orEmpty(),
+                        version = ref.orEmpty()
+                    ),
+                    definitionFilePath = VersionControlSystem.getPathInfo(definitionFile).path,
+                    declaredLicenses = sortedSetOf(),
+                    vcs = VcsInfo.EMPTY,
+                    vcsProcessed = processProjectVcs(workingDir, VcsInfo.EMPTY),
+                    scopes = sortedSetOf(),
+                    homepageUrl = ""
+                ),
+                packages = parseCarthageDependencies(definitionFile)
+            )
+        )
+    }
+
+    /**
+     * As the Carthage.resolved file does not provide any project information, trying to retrieve some from VCS.
+     */
+    private fun getProjectInfoFromVcs(workingDir: File): Triple<String?, String?, String?> {
```

Comment:
I think the triple makes it a bit harder for the reader (of the signature) to figure out what is actually returned.
Please introduce a class for the return type to address that. How about `data class ProjectInfo(...)`